pkg/proxy/ipvs/proxier.go

```diff
@@ -1384,16 +1387,30 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 	return nil
 }
 
-func (proxier *Proxier) cleanLegacyService(atciveServices map[string]bool, currentServices map[string]*utilipvs.VirtualServer) {
+func (proxier *Proxier) cleanLegacyService(activeServices map[string]bool, currentServices map[string]*utilipvs.VirtualServer) {
 	unbindIPAddr := sets.NewString()
-	for cS := range currentServices {
-		if !atciveServices[cS] {
-			svc := currentServices[cS]
-			err := proxier.ipvs.DeleteVirtualServer(svc)
-			if err != nil {
-				glog.Errorf("Failed to delete service, error: %v", err)
+	for cs := range currentServices {
+		svc := currentServices[cs]
+		if _, ok := activeServices[cs]; !ok {
+			// This service was not processed in the latest sync loop so before deleting it,
+			// make sure it does not fall within an excluded CIDR range.
+			okayToDelete := true
+			for _, excludedCIDR := range proxier.excludeCIDRs {
+				_, n, err := net.ParseCIDR(excludedCIDR)
```

Comment:
Why are we validating this again? Isn't it supposed to be validated during configuration check?

Reply:
If we can't ignore the error here, then I'd say we can skip configuration validation completely and just have the validation happen here.

Reply:
The intent was not validation but rather to convert the string representation of the CIDR to net struct. I guess I could remove the error check since we know from the previous validation that no error is possible? Does that sound okay?

Reply:
SGTM.

Reply:
Done